Change to rollupjs and extend interoperability with Element client (#6)
owncloud/ocis-element: element-messenger/README.md

# Simple Element Messenger Integration

Element is a client solution for the popular Matrix chat protocol.
This extension has two parts, the first one lets you embed a self-hosted Elements client into your ownCloud environment.

> The second is a widget for usage within Elements, that allows to directly share files with within chat rooms.

Reviewer: ```suggestion
The second is an Elements widget, which allows to directly share files inside chat rooms.
```

Author: > [...]One more thing - in the latest version of file picker, we're also including library build. That means it can be imported into Vue projects directly without using the web component. Maybe that would be something of an interest for you?

Yes, using the library build should be better. I think in a future release we can address this

You will need a self hosted Element client as the official client does not allow for frame based embedding. Just follow the given setup instructions of the Element client and set your webserver to emit correct X-Frame-Options headers for your environment. See https://developer.mozilla.org/de/docs/Web/HTTP/Headers/X-Frame-Options

Furthermore extend your main config.json with a configuration that looks like the following:

    "external_apps": [
      {
        "id": "element-messenger",
        "path": "web-app-element-messenger",
        "config": {
          "server": "<path to your Element instance>"
        }
      }
    ]

To deploy the app just deploy the contents of the top level dist folder in your instance of ownCloud Web to the **js** folder.

## Deployment of Elements file picker widget
Please deploy the contents (the widget) of the dist folder within the element-file-picker folder on your server.
You can deploy it anywhere you like. After deployment you will need to create a file-picker.config.json file if you want to use the file picker widget in Matrix.
Look here for further information: https://owncloud.github.io/integration/file_picker/installation/

> Please put the finished as file-picker-config.json in the same folder as the widget.

Reviewer: ```suggestion
Please, put the finished config file as file-picker-config.json in the same folder as the widget.
```
